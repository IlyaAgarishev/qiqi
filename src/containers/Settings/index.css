.settings {
  position: relative;
  top: -170px;
  left: 80px;
  width: 300px;
  /* height: 150px; */
  background: #0091d7;
  transition: all 0.3s ease;
  opacity: 0;
  padding: 10px;
}

.wordsLimit-settings {
  display: flex;
  flex-direction: column;
  background: #019fee;
  padding: 15px;
  /* border-radius: 10px; */
}

.wordsLimit-settings-title {
  font-weight: 500;
  color: white;
}

.wordsLimit-cells {
  display: flex;
  margin-top: 10px;
}

.wordsLimit-cell {
  padding: 20px;
  font-size: 22px;
  font-weight: 900;
  color: white;
  background: #0091d7;
  margin-right: 10px;
  cursor: pointer;
  transition: all 0.3s ease;
  border-radius: 10px;
}

.wordsLimit-cell:hover {
  color: #0091d7;
  background: white;
}
