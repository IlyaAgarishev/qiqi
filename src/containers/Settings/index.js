/* global chrome */

import React from 'react';
import './index.css';

class Settings extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    if (this.props.open) {
      this.settings.style.display = 'block';
    } else {
      this.settings.style.display = 'none';
    }

    this.enableSingleWordsLimitStyle(this.wordsLimitCells.children);
  }

  disableAllWordsLimitStyle = children => {
    for (let i = 0; i < children.length; i++) {
      children[i].style.color = 'white';
      children[i].style.background = '#0091d7';
    }
  };

  enableSingleWordsLimitStyle = children => {
    chrome.storage.sync.get(['wordsLimit'], storageData => {
      for (let i = 0; i < children.length; i++) {
        let synthIndex = i + 1;
        if (storageData.wordsLimit == synthIndex * 10) {
          children[i].style.color = '#0091d7';
          children[i].style.background = 'white';
        }
      }
    });
  };

  // setWordsLimitSettings = (number, children) => {
  //   this.disableAllWordsLimitStyle(children);
  //   this.props.setWordsLimitStorage(number);
  //   this.enableSingleWordsLimitStyle(children);
  // };

  shouldComponentUpdate(props) {
    if (props.open) {
      this.settings.style.display = 'block';
      setTimeout(() => {
        this.settings.style.opacity = '1';
      }, 10);
    } else {
      this.settings.style.opacity = '0';
      setTimeout(() => {
        this.settings.style.display = 'none';
      }, 300);
    }
    return true;
  }

  render() {
    return (
      <div className="settings" ref={ref => (this.settings = ref)}>
        <div className="wordsLimit-settings">
          <div className="wordsLimit-settings-title">Количество слов в тесте :</div>
          <div className="wordsLimit-cells" ref={ref => (this.wordsLimitCells = ref)}>
            <div
              className="wordsLimit-cell"
              onClick={() => {
                this.disableAllWordsLimitStyle(this.wordsLimitCells.children);
                this.props.setWordsLimitStorage(10);
                this.enableSingleWordsLimitStyle(this.wordsLimitCells.children);
              }}
            >
              10
            </div>
            <div
              className="wordsLimit-cell"
              onClick={() => {
                this.disableAllWordsLimitStyle(this.wordsLimitCells.children);
                this.props.setWordsLimitStorage(20);
                this.enableSingleWordsLimitStyle(this.wordsLimitCells.children);
              }}
            >
              20
            </div>
            <div
              className="wordsLimit-cell"
              onClick={() => {
                this.disableAllWordsLimitStyle(this.wordsLimitCells.children);
                this.props.setWordsLimitStorage(30);
                this.enableSingleWordsLimitStyle(this.wordsLimitCells.children);
              }}
            >
              30
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Settings;
