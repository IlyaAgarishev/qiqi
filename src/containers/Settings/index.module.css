.settings {
  position: relative;
  top: -170px;
  left: 80px;
  width: 300px;
  background: #0091d7;
  transition: all 0.3s ease;
  opacity: 1;
  padding: 10px;
  border-radius: 10px;
  display: block;
}

.settingsHide {
  display: none;
}

.wordsLimitSettings {
  display: flex;
  flex-direction: column;
  padding: 15px;
}

.wordsLimitSettingsTitle {
  font-weight: 500;
  font-size: 16px;
  color: white;
}

.wordsLimitCells {
  display: flex;
  margin-top: 10px;
}

.wordsLimitCell {
  padding: 20px;
  font-size: 22px;
  font-weight: 900;
  color: white;
  margin-right: 10px;
  cursor: pointer;
  transition: all 0.3s ease;
  border-radius: 10px;
}
