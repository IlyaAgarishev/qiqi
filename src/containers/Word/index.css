.word {
  margin-right: 10px;
  background: white;
  margin-bottom: 8px;
  padding: 10px;
  font-size: 16px;
}

.word-fat {
  font-weight: bold;
}
