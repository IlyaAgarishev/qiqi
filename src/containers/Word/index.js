import React from 'react';
import './index.css';
import CloseBtn from '../../img/cancel.svg';

class Word extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div
        className="word"
        onClick={() => {
          this.props.deleteWordFromTest(this.props.index);
        }}
      >
        <div className="word-fat">{this.props.originalWord}</div>
        <div className="word-translation">{this.props.translatedWord}</div>
      </div>
    );
  }
}

export default Word;
