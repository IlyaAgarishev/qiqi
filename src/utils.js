/* global chrome */

export const wordEndingGenerator = (wordsLimit, dictionaryLength) => {
  let wordsLeft = wordsLimit - dictionaryLength;
  if (
    (wordsLeft >= 5 && wordsLeft <= 14) ||
    (wordsLeft % 10 >= 5 && wordsLeft % 10 <= 9) ||
    wordsLeft % 10 === 0
  ) {
    return `${wordsLeft} слов`;
  } else if (wordsLeft >= 2 && wordsLeft <= 4) {
    return `${wordsLeft} слова`;
  } else if (wordsLeft === 1 || (wordsLeft % 10 >= 1 && wordsLeft % 10 <= 4)) {
    return `${wordsLeft} слово`;
  }
};

export const wordsLimitGet = callback => {
  chrome.storage.sync.get(["wordsLimit"], storageData => {
    callback(storageData.wordsLimit);
  });
};

export const wordsLimitSet = number => {
  chrome.storage.sync.set({ wordsLimit: number });
};

// export const dictionaryGet = callback => {
//   chrome.storage.sync.get(["dictionary"], storageData => {
//     callback(storageData.dictionary);
//   });
// };

// export const wordsLimitSet = callback => {
//   chrome.storage.sync.get(["wordsLimit"], storageData => {
//     callback(storageData.wordsLimit);
//   });
// };

// chrome.storage.sync.set({ wordsLimit: number }, () => {
//   setWordsLimitState();
// });
