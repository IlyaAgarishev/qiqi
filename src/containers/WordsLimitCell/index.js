import React from "react";
import styles from "./index.module.css";
import PropTypes from "prop-types";
import { wordsLimitSet } from "../../utils";

const WordsLimitCell = props => {
  const { index, setWordsLimit, certainCell, setCertainCell } = {
    ...props
  };
  return (
    <div
      className={
        certainCell === index
          ? [styles.wordsLimitCell, styles.white].join(" ")
          : styles.wordsLimitCell
      }
      onClick={() => {
        setWordsLimit(index);
        setCertainCell(index);
        wordsLimitSet(index);
      }}
    >
      {index}
    </div>
  );
};

WordsLimitCell.propTypes = {
  index: PropTypes.number.isRequired,
  setWordsLimit: PropTypes.func.isRequired,
  certainCell: PropTypes.number.isRequired,
  setCertainCell: PropTypes.func.isRequired
};

export default WordsLimitCell;
