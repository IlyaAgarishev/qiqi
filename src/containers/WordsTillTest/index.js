import React from "react";
import styles from "./index.module.css";
import PropTypes from "prop-types";
import { wordEndingGenerator } from "../../utils";

const WordsTillTest = props => {
  const { wordsLimit, dictionaryLength } = { ...props };
  const wordsLeft = wordsLimit - dictionaryLength;
  return (
    <div className={styles.wordsTillTest}>
      Вам осталось набрать
      <div className={styles.numberOfWords}>
        {wordEndingGenerator(wordsLeft)}
      </div>
      чтобы начать тест
    </div>
  );
};

WordsTillTest.propTypes = {
  wordsLimit: PropTypes.number.isRequired,
  dictionaryLength: PropTypes.number.isRequired
};

export default WordsTillTest;
