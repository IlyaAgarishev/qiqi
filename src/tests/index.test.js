import React from "react";
import { mount, render, shallow } from "enzyme";
import { wordEndingGenerator } from "../utils";
import WordsTillTest from "../containers/WordsTillTest";
import WordsLimitCell from "../containers/WordsLimitCell";
import Settings from "../containers/Settings";
import Word from "../containers/Word";

// functions-tools for testing
const digitsArratGenerator = conditions => {
  let digitsArray = [];
  for (let i = 0; i <= 30; i++) {
    if (conditions(i)) {
      digitsArray.push(i);
    }
  }
  return digitsArray;
};

const conditionsForWord_slov = i => {
  return (i % 10 >= 5 && i % 10 <= 9) || i % 10 === 0 || (i >= 11 && i <= 14);
};

const conditionsForWord_slova = i => {
  return i % 10 >= 2 && i % 10 <= 4 && i !== 12 && i !== 13 && i !== 14;
};

const conditionsForWord_slovo = i => {
  return i % 10 === 1 && i !== 11;
};

// Functions testing

test("wordEndingGenerator returns string", () => {
  expect(typeof wordEndingGenerator(10)).toBe("string");
});

test("wordEndingGenerator returns word 'slov", () => {
  digitsArratGenerator(conditionsForWord_slov).forEach(el => {
    expect(wordEndingGenerator(el)).toBe(`${el} слов`);
  });
});

test("wordEndingGenerator returns word 'slova", () => {
  digitsArratGenerator(conditionsForWord_slova).forEach(el => {
    expect(wordEndingGenerator(el)).toBe(`${el} слова`);
  });
});

test("wordEndingGenerator returns word 'slovo", () => {
  digitsArratGenerator(conditionsForWord_slovo).forEach(el => {
    expect(wordEndingGenerator(el)).toBe(`${el} слово`);
  });
});

// Components testing

test("WordsTillTest renders right string", () => {
  const wordsLimit = 30;
  for (let i = 0; i <= wordsLimit; i++) {
    const component = mount(
      <WordsTillTest wordsLimit={wordsLimit} dictionaryLength={i} />
    );
    const wordsLeft = wordsLimit - i;
    expect(component.find(".numberOfWords").text()).toBe(
      wordEndingGenerator(wordsLeft)
    );
  }
});

test("WordsLimitCell renders right string", () => {
  const index = 10;
  const component = mount(
    <WordsLimitCell index={index} setWordsLimit={jest.fn()} wordsLimit={30} />
  );
  expect(component.find(".wordsLimitCell").text()).toBe(`${index}`);
});

test("Settings renders right string", () => {
  const component = mount(
    <Settings
      setWordsLimit={jest.fn()}
      wordsLimit={30}
      setOpenSettings={jest.fn()}
    />
  );
  const numbersArray = [10, 20, 30];
  numbersArray.forEach((number, index) => {
    expect(
      component
        .find(".wordsLimitCell")
        .at(index)
        .text()
    ).toBe(`${number}`);
  });
});

test("Word renders right string", () => {
  const props = {
    index: 1,
    originalWord: "wow",
    translatedWord: "воу",
    dictionary: [],
    setDictionary: jest.fn()
  };
  const component = mount(<Word {...props} />);
  expect(component.find(".originalWord").text()).toBe(props.originalWord);
  expect(component.find(".translatedWord").text()).toBe(props.translatedWord);
});
