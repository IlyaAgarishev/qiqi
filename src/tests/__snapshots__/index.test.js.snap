// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots 1`] = `
<div
  className="wordsTillTest"
>
  Вам осталось набрать
  <div
    className="numberOfWords"
  >
    20 слов
  </div>
  чтобы начать тест
</div>
`;

exports[`Snapshots 2`] = `
<div
  className="wordsLimitCell"
  onClick={[Function]}
>
  10
</div>
`;

exports[`Snapshots 3`] = `
<div>
  <div
    className="closeSettings"
    onClick={[Function]}
  />
  <div
    className="settings"
  >
    <div
      className="wordsLimitSettings"
    >
      <div
        className="wordsLimitSettingsTitle"
      >
        Количество слов в тесте :
      </div>
      <div
        className="wordsLimitCells"
      >
        <WordsLimitCell
          index={10}
          key="0"
          setWordsLimit={[MockFunction]}
          wordsLimit={30}
        />
        <WordsLimitCell
          index={20}
          key="1"
          setWordsLimit={[MockFunction]}
          wordsLimit={30}
        />
        <WordsLimitCell
          index={30}
          key="2"
          setWordsLimit={[MockFunction]}
          wordsLimit={30}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshots 4`] = `
<div
  className="word"
  onClick={[Function]}
>
  <strong>
    wow
  </strong>
  <div>
    воу
  </div>
</div>
`;

exports[`Snapshots 5`] = `
<div
  className="dictionary transformUp"
>
  <div
    className="dictionaryTools"
  >
    <div
      className="closeBtn"
      onClick={[Function]}
    >
      <img
        alt=""
        className="closeBtnImg"
        src="cancel.svg"
      />
    </div>
    <div
      className="clearDictionary"
      onClick={[Function]}
    >
      Очистить
    </div>
  </div>
  <div
    className="words"
  >
    <Word
      dictionary={
        Array [
          Object {
            "translation": "кот",
            "word": "cat",
          },
          Object {
            "translation": "пес",
            "word": "dog",
          },
          Object {
            "translation": "лев",
            "word": "lion",
          },
        ]
      }
      index={0}
      key="0"
      originalWord="cat"
      setDictionary={[MockFunction]}
      translatedWord="кот"
    />
    <Word
      dictionary={
        Array [
          Object {
            "translation": "кот",
            "word": "cat",
          },
          Object {
            "translation": "пес",
            "word": "dog",
          },
          Object {
            "translation": "лев",
            "word": "lion",
          },
        ]
      }
      index={1}
      key="1"
      originalWord="dog"
      setDictionary={[MockFunction]}
      translatedWord="пес"
    />
    <Word
      dictionary={
        Array [
          Object {
            "translation": "кот",
            "word": "cat",
          },
          Object {
            "translation": "пес",
            "word": "dog",
          },
          Object {
            "translation": "лев",
            "word": "lion",
          },
        ]
      }
      index={2}
      key="2"
      originalWord="lion"
      setDictionary={[MockFunction]}
      translatedWord="лев"
    />
  </div>
</div>
`;
